packages/react-components/react-swatch-picker-preview/src/components/SwatchPicker/useSwatchPicker.ts

```diff
@@ -11,24 +14,48 @@ import type { SwatchPickerProps, SwatchPickerState } from './SwatchPicker.types'
  * @param props - props from this instance of SwatchPicker
  * @param ref - reference to root HTMLElement of SwatchPicker
  */
-export const useSwatchPicker_unstable = (
+export const useSwatchPicker_unstable = <T>(
   props: SwatchPickerProps,
   ref: React.Ref<HTMLDivElement>,
 ): SwatchPickerState => {
-  return {
-    // TODO add appropriate props/defaults
+  const { defaultSelectedValue, role, onSelectionChange, selectedValue, size, shape, ...rest } = props;
+  const focusAttributes = useArrowNavigationGroup({
+    circular: true,
+    axis: 'both',
+    memorizeCurrent: true,
+  });
+
+  const [selectedSwatch, setSelectedSwatch] = useControllableState({
+    state: selectedValue,
+    defaultState: defaultSelectedValue,
+    initialState: '',
+  });
```

Comment:
```suggestion
  const [selectedValue, setSelectedValue] = useControllableState({
    state: props.selectedValue,
    defaultState: props.defaultSelectedValue,
    initialState: '',
  });
```

This should help to avoid naming conflicts and keep naming consistent